### DISCUSSION TITLE
What special routes are there in remix?

### DISCUSSION THREAD

**Discussion by a-h** [remix-run/remix#10286] [Q&A]
I'm trying to get Remix to work alongside another application on the same HTTP host.

I set it up so that the prefixes `/api`, and `/_` go to my other application, but then I noticed that Remix sends requests to `_root.data?index` which I wasn't expecting.

Is there a list of the paths that Remix will use, or a way to differentiate them, or is this the only one?

**Comment by sergiodxa** [comment]
With Single Fetch, when doing a client-side navigation let's say to `/foo` Remix will send a request to `/foo.data` to get the data

without Single Fetch, when doing a client-side navigation let's say to `/foo` Remix will send a request to `/foo?_route=routeId` where `routeId` is the id of the route's data it wants to fetch.

For the root route, with Single Fetch that still loads from `/foo.data`, but without Single Fetch it will load from `/foo?_route=root`

Without Single Fetch there's also a case where it adds `?index`, that's used for POST requests to root or an _index route.

**Comment by a-h** [comment]
Thanks, it sounds like the `index` is given the name `/_root` so I guess that's the only edge case I need to think about, unless I add an `api` route to my Remix app, in which case that's my own fault. 😁
